x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/documents.tsx

```diff
@@ -71,26 +54,31 @@ export const SearchIndexDocuments: React.FC = () => {
       : `${CONNECTORS_ACCESS_CONTROL_INDEX_PREFIX}${indexName}`;
   const mappingLogic = mappingsWithPropsApiLogic(indexToShow);
   const documentLogic = searchDocumentsApiLogic(indexToShow);
-
   const { makeRequest: getDocuments } = useActions(documentLogic);
   const { makeRequest: getMappings } = useActions(mappingLogic);
   const { data, status, error } = useValues(documentLogic);
   const { data: mappingData, status: mappingStatus } = useValues(mappingLogic);
 
-  const docs = data?.results?.hits.hits ?? [];
```

Comment:
I don't see where the data response was changed to require this change for where to access documents.

I see that `searchDocumentsApiLogic` used to return the response directly, but now adds pagination:

```typescript
return {
    meta: pageToPagination(response.results?._meta?.page),
    results: response.results?.data,
  };
```

But that doesn't change where the documents are defined away from `.hits.hits`, what am I missing here?

Reply:
The magic happens in [fetchWithPagination](https://github.com/elastic/kibana/blob/d6cd3f446faa078c610f9a47ef0b3d7586da7192/packages/kbn-search-index-documents/utils/fetch_with_pagination.ts) 


https://github.com/elastic/kibana/blob/d6cd3f446faa078c610f9a47ef0b3d7586da7192/packages/kbn-search-index-documents/utils/fetch_with_pagination.ts#L36-L44